### DISCUSSION TITLE
Cross-Platform Website Launcher with Avalonia

### DISCUSSION THREAD

**Discussion by Rushikesh3009** [AvaloniaUI/Avalonia#13640] [Q&A]
I am looking for a universal code that works on all platforms (Windows, Android, and iOS) using Avalonia. The code should allow me to open any website using a URL, For example "http://www.google.com/" .

**Comment by thevortexcloud** [comment]
I don't think there is a built in way to do this. But it looks like Avalonia Essentials can do it possibly.

https://github.com/AvaloniaUI/Avalonia.Essentials

It's a port of Xamarin essentials so most of the documentation is just as applicable.

https://learn.microsoft.com/en-us/dotnet/api/xamarin.essentials.launcher.openasync?view=xamarin-essentials-android

There is a bunch of examples here too:

https://github.com/AvaloniaUI/Avalonia.Essentials/blob/00bf90d8cbdf55cb48fef87be693ed9d82dc9b3b/samples/Essentials.Sample/ViewModel/LauncherViewModel.cs#L52

**Reply by Rushikesh3009** [reply]
I added `using Microsoft.Maui.ApplicationModel;` to the header, but I am getting the following error.

![image](https://github.com/AvaloniaUI/Avalonia/assets/149040605/0576ed8c-2c75-46f9-9efd-9b47e8144787)

I don't know where I am going wrong. Could you please provide me with the code for it?

**Reply by thevortexcloud** [reply]
I am not sure how you got MAUI out of this? I gave a link for Avalonia Essentials which is a different thing entirely. I don't think there is a package available for Avalonia essentials though so you may just need to build it yourself. I am not sure, I have never used it so someone else may be able to tell you how to get it if you don't want to build it. You can use MAUI with Avalonia (it's sort of officially supported) but you may be unintentionally limiting yourself as MAUI does not support all of the same platforms as Avalonia does. Not that I think it matters much here.

In any case that's not an actual error. It's a warning telling you that you have platform specific code that can be technically executed on a platform it's not designed for. In most cases you can ignore it if you completely sure that you will not call it on the unsupported platforms. You can also guard it with the various `OperatingSystem` methods which may make the warning go away. In this instance it seems to be telling you that older versions of Mac/iOS can't use this method. But newer ones can.

EDIT: Apparently you do need to build Avalonia Essentials yourself.

https://github.com/AvaloniaUI/Avalonia/discussions/12891

**Reply by Rushikesh3009** [reply]
Ohh! Thank you for your Support.

And  sorry for misunderstanding.

**Comment by robloo** [comment]
As noted there, is presently no built-in Avalonia method of doing this. However, it's going to be needed soon(ish) for the `HyperlinkButton` so it will happen eventually.

More discussion, as well as some existing code, is here:
 - https://github.com/AvaloniaUI/Avalonia/pull/9385#issuecomment-1308175710

**Reply by thevortexcloud** [reply]
There is now a PR https://github.com/AvaloniaUI/Avalonia/pull/14320
